spec/controllers/users_controller_spec.rb

```diff
@@ -250,6 +250,56 @@
         expect(response).to render_template(:edit)
       end
     end
+
+    context 'send confirmation mails only on email change and signup' do
+      it 'should receive a confirmation mail on email change' do
+        @request.env["devise.mapping"] = Devise.mappings[:user]
+        user = User.create! valid_attributes
+        sign_in user
+        ActionMailer::Base.deliveries.clear
+
+        old_mail = user.email
+        new_mail = 'my_new_mail@example.com'
+        new_attributes = { email: new_mail,
+                           current_password: user.password }
+        put :update, params: { id: user.to_param, user: new_attributes }
+
+        expect(ActionMailer::Base.deliveries.length).to eq(2)
+        notification_mail = ActionMailer::Base.deliveries.first
+        notification_mail_body = Capybara.string(notification_mail.body.encoded)
+        expect(notification_mail.to.length).to eq(1) #one receiver
```

Comment:
I think this should be part of a mailer spec, you should only verify that the right mailer is invoked.